test/e2e/framework/framework.go

```diff
@@ -159,205 +135,38 @@ func (f *Framework) GetDefaultScyllaCluster() *scyllav1.ScyllaCluster {
-
-	// Create a service account token Secret for the user ServiceAccount.
-	userSATokenSecret, err := f.KubeAdminClient().CoreV1().Secrets(ns.Name).Create(ctx, &corev1.Secret{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: ServiceAccountTokenSecretName,
-			Annotations: map[string]string{
-				corev1.ServiceAccountNameKey: userSA.Name,
-			},
-		},
-		Type: corev1.SecretTypeServiceAccountToken,
-	}, metav1.CreateOptions{})
-	o.Expect(err).NotTo(o.HaveOccurred())
-
-	By("Waiting for service account token Secret %q in namespace %q.", userSATokenSecret.Name, userSATokenSecret.Namespace)
-	ctxUserSATokenSecret, ctxUserSATokenSecretCancel := context.WithTimeout(ctx, serviceAccountTokenSecretWaitTimeout)
-	defer ctxUserSATokenSecretCancel()
-	userSATokenSecret, err = WaitForServiceAccountTokenSecret(ctxUserSATokenSecret, f.KubeAdminClient().CoreV1(), userSATokenSecret.Namespace, userSATokenSecret.Name)
-	o.Expect(err).NotTo(o.HaveOccurred())
-	o.Expect(userSATokenSecret.Data).To(o.HaveKey(corev1.ServiceAccountTokenKey))
-
-	token := userSATokenSecret.Data[corev1.ServiceAccountTokenKey]
-	o.Expect(token).NotTo(o.BeEmpty())
-
-	// Create a restricted client using the user SA.
-	f.clientConfig = restclient.AnonymousClientConfig(f.AdminClientConfig())
-	f.clientConfig.BearerToken = string(token)
-
-	// Wait for default ServiceAccount.
-	By("Waiting for default ServiceAccount in namespace %q.", ns.Name)
-	ctxSa, ctxSaCancel := context.WithTimeout(ctx, serviceAccountWaitTimeout)
-	defer ctxSaCancel()
-	_, err = WaitForServiceAccount(ctxSa, f.KubeAdminClient().CoreV1(), ns.Name, "default")
-	o.Expect(err).NotTo(o.HaveOccurred())
-
-	// Waits for the configmap kube-root-ca.crt containing CA trust bundle so that pods do not have to retry mounting
-	// the config map because it creates noise and slows the startup.
-	By("Waiting for kube-root-ca.crt in namespace %q.", ns.Name)
-	_, err = controllerhelpers.WaitForConfigMapState(
-		ctx,
-		f.KubeClient().CoreV1().ConfigMaps(ns.Name),
-		"kube-root-ca.crt",
-		controllerhelpers.WaitForStateOptions{},
-		func(configMap *corev1.ConfigMap) (bool, error) {
-			return true, nil
-		},
-	)
-	o.Expect(err).NotTo(o.HaveOccurred())
-}
+func (f *Framework) beforeEach() {
+	ctx, ctxCancel := context.WithCancel(context.Background())
+	defer ctxCancel()
 
-func (f *Framework) deleteNamespace(ctx context.Context, ns *corev1.Namespace) {
-	By("Destroying namespace %q.", ns.Name)
-	var gracePeriod int64 = 0
-	var propagation = metav1.DeletePropagationForeground
-	err := f.KubeAdminClient().CoreV1().Namespaces().Delete(
-		ctx,
-		ns.Name,
-		metav1.DeleteOptions{
-			GracePeriodSeconds: &gracePeriod,
-			PropagationPolicy:  &propagation,
-			Preconditions: &metav1.Preconditions{
-				UID: &ns.UID,
-			},
-		},
-	)
-	o.Expect(err).NotTo(o.HaveOccurred())
+	var wg sync.WaitGroup
+	defer wg.Wait()
 
-	// We have deleted only the namespace object but it is still there with deletionTimestamp set.
+	for _, c := range f.clusters {
```

Comment:
auto namespace only in the default one

Reply:
https://github.com/scylladb/scylla-operator/pull/1881#discussion_r1606594825